src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -71,36 +159,78 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
                 return;
             }
 
+            // NOTE: Values in TagHelperOutput.Attributes are already HtmlEncoded
+
             var content = new StringBuilder();
+            
+            if (modeResult.Mode == Mode.Fallback && string.IsNullOrEmpty(HrefInclude))
+            {
+                // No globbing to do, just build a <link /> tag to match the original one in the source file
+                BuildLinkTag(output.Attributes, content);
+            }
+            else
+            {
+                BuildGlobbedLinkTags(output.Attributes, content);
+            }
 
-            // NOTE: Values in TagHelperOutput.Attributes are already HtmlEncoded
+            if (modeResult.Mode == Mode.Fallback)
+            {
+                BuildFallbackBlock(content);
+            }
 
-            // We've taken over rendering here so prevent the element rendering the outer tag
+            // We've taken over rendering so prevent the element rendering the outer tag
             output.TagName = null;
+            output.Content = content.ToString();
+        }
 
-            // Rebuild the <link /> tag that loads the primary stylesheet
-            content.Append("<link ");
-            foreach (var attribute in output.Attributes)
+        private void BuildGlobbedLinkTags(IDictionary<string, string> attributes, StringBuilder content)
+        {
+            // Build a <link /> tag for each matched href as well as the original one in the source file
+            string staticHref;
+            attributes.TryGetValue("href", out staticHref);
+
+            var hrefs = GlobbingUtil.BuildUrlList(staticHref, HrefInclude, HrefExclude);
+
+            foreach (var href in hrefs)
             {
-                content.AppendFormat(CultureInfo.InvariantCulture, "{0}=\"{1}\" ", attribute.Key, attribute.Value);
+                attributes["href"] = WebUtility.HtmlEncode(href);
```

Comment:
Since you pass inthe `output.Attributes` here you should probably set the `href` back to the initial value in the case someone else targets the `<link>` tag with a `TagHelper`. That way we're not using the `output.Attributes` as a temporary store and inadvertently nuke its validity.